### PR TITLE
Hash IDs in URLs - resolves #659

### LINKED ISSUE #659: No ID numbers in URLs
Use something like [hashids](http://hashids.org/python/) to obfuscate the ID numbers

### PR DESCRIPTION
**Pull request by knrafto**
Not sure where to put the converter class. utils.py maybe?

### CODE REVIEW

File: server/__init__.py [hunk 1]
```diff
@@ -20,6 +22,17 @@
     debug_toolbar
 )
 
+class HashidConverter(BaseConverter):
+    # ID hashing configuration.
+    # DO NOT CHANGE ONCE THE APP IS PUBLICLY AVAILABLE. You will break every
```
Comment (Sumukh) [inline server/__init__.py]: Should we add a test to make sure that 
 - the hashid converter exists
 - that the config hasn't changed (by encoding/decoding a value) and comparing it's value.

**Comment by Sumukh** [comment]
I agree that utils.py is a better place to put the class. The templates will have to use this to convert some IDs. (or we could make it a model method)

File: server/utils.py [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
-    return models.Assignment.query.filter_by(name=name).one_or_none()
+    def to_python(self, value):
+        numbers = self.hashids.decode(value)
+        if len(numbers) != 1:
+            raise ValidationError('Could not decode hash {} into ID'.format(value))
+        return numbers[0]
 
-def course_by_name(name):
-    return models.Course.query.filter_by(offering=name).one_or_none()
+    def to_url(self, value):
```
Comment (knrafto) [inline server/utils.py]: This doesn't seem to be used anywhere

**Comment by knrafto** [comment]
LGTM, merging

**Pull request merged**